### PR TITLE
fix: indicator position of error message

### LINKED ISSUE #3766: Fail message annotation points to the wrong line
### Describe the bug

Console annotations of test failures -- specifically the red `^` -- often appears on the wrong line and at the wrong character. 
Capture

In my real code I was tricked into thinking the earlier assertions were passing when they were not.

The behavior seems to be a function of the number of lines in the test file. For example, in my repro, note the ~10 lines preceding the test. If these are reduced to 1 or 2 lines, the annotation ends up in the correct place. 

### Reproduction

```
git clone https://github.com/davemfish/repro-vitest-annotation
cd repro-vitest-annotation
yarn test
```

### System Info

```shell
λ npx envinfo --system --npmPackages '{vitest,@vitest/*,vite,@vitejs/*}' --binaries --browsers
npx: installed 1 in 2.539s

  System:
    OS: Windows 10 10.0.19045
    CPU: (12) x64 AMD Ryzen 5 1600 Six-Core Processor
    Memory: 12.73 GB / 31.95 GB
  Binaries:
    Node: 18.14.2 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.11 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 6.14.11 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 114.0.5735.199
    Edge: Spartan (44.19041.1266.0), Chromium (114.0.1823.79)
    Internet Explorer: 11.0.19041.1566
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by fenghan34**
### Description

Fixes #3766 

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

**Review by sheremet-va — CHANGES_REQUESTED** [review]
Please, add a test. And don't remove the PR description

**Comment by fenghan34** [comment]
It seems that I can't push CRLF formatted files. How can I add tests for this, Do you have any suggestions? @sheremet-va

**Comment by sheremet-va** [comment]
> It seems that I can't push CRLF formatted files. How can I add tests for this, Do you have any suggestions? @sheremet-va

I don't know what you mean by "I can't push CRLF formatted files". Is it an eslint rule? You can disable it in a single file.

**Comment by sheremet-va** [comment]
You can also just test a single function call.

**Comment by fenghan34** [comment]
>  I don't know what you mean by "I can't push CRLF formatted files"

It's not eslint rule. This issue relates to CRLF format, see https://github.com/vitest-dev/vitest/issues/3766#issuecomment-1637145601. 

To add a test, I need to create a CRLF format test file, but we have the `.gitattributes` config, and it tells git to normalize all line endings to LF.

**Comment by fenghan34** [comment]
> You can also just test a single function call.

What do you mean by "test a single function call"?

**Comment by fenghan34** [comment]
Can I add a line to `.gitattributes`? Seems like we can specify a single file in CRLF format

**Comment by sheremet-va** [comment]
> Can I add a line to `.gitattributes`? Seems like we can specify a single file in CRLF format

Sure

**Review by sheremet-va — APPROVED** [review]

File: test/reporters/fixtures/indicator-position.test.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable no-multiple-empty-lines */
```
Comment (sheremet-va) [inline test/reporters/fixtures/indicator-position.test.ts:1]: I think this file should be javascript because esbuild truncates all empty lines.

**Review by sheremet-va — CHANGES_REQUESTED** [review]

**Review by sheremet-va — APPROVED** [review]

**Comment by sheremet-va** [comment]
LGTM 👍

**Comment by mcavigelli** [comment]
Thanks a lot for the fix!

**Pull request merged**